packages/lib/modules/lbp/steps/useCreateLbpSteps.tsx

```diff
@@ -36,20 +44,79 @@ export function useCreateLbpSteps() {
   }
 
   const signPermit2Step = useSignPermit2InitializeStep({ initPoolInput })
+  // If user chooses setting to not use signatures, use these approval txs
+  const { steps: permit2ApprovalSteps, isLoading: isLoadingPermit2ApprovalSteps } =
+    usePermit2ApprovalSteps({
+      chain: selectedChain,
+      approvalAmounts: initAmounts,
+      actionType: 'InitializePool',
+      enabled: !shouldUseSignatures,
+    })
+
   const initLbpStep = useInitializeLbpStep({ initPoolInput })
-  const isLoadingSteps = !initAmounts.length || isLoadingTokenApprovalSteps || !signPermit2Step
+
+  const isLoadingSteps =
+    !initAmounts.length ||
+    !signPermit2Step ||
+    isLoadingTokenApprovalSteps ||
+    isLoadingPermit2ApprovalSteps
+
+  const steps = getApprovalAndCreateSteps({
+    shouldUseSignatures,
+    signPermit2Step,
+    permit2ApprovalSteps,
+    initLbpStep,
+    tokenApprovalSteps,
+    shouldBatchTransactions,
+  })
 
   return {
     isLoadingSteps,
-    steps: [
-      createLbpStep,
-      ...tokenApprovalSteps,
-      ...(signPermit2Step ? [signPermit2Step] : []),
-      initLbpStep,
-    ],
+    steps: [createLbpStep, ...steps],
   }
 }
 
+function getApprovalAndCreateSteps({
```

Comment:
Isn't this the same / similar function that we have for add liquidity steps? can we somehow generalize them to be shared? (maybe even remove liquidity steps to even though the code is a little different there)

Reply:
yeah thats a good point! will work on the refactor